v2/http-middleware/ratelimit/ratelimit.go

```diff
@@ -0,0 +1,115 @@
+package ratelimit
+
+import (
+	"fmt"
+	"github.com/SKF/go-utility/v2/http-middleware/util"
+	"github.com/gomodule/redigo/redis"
+	"github.com/gorilla/mux"
+	"net/http"
+	"time"
+)
+
+type Store interface {
+	Incr(string) (int, error)
+}
+
+type EndpointConfig struct {
+	path   Request
+	config []Config
+}
+
+type Config struct {
+	requestPerMinute int
+	getKeyFunc       func(req *http.Request) (string, error)
+}
+
+type Request struct {
+	method string
+	path   string
+}
+
+type Limiter struct {
+	store   Store
+	configs map[Request][]Config
+}
+
+func SetStore(s Store) Limiter {
+	return Limiter{
+		store:   s,
+		configs: map[Request][]Config{},
+	}
+}
+
+func (s *Limiter) Configure(config EndpointConfig) {
+	s.configs[config.path] = config.config
+}
+
+type redisStore struct {
+	url string
+}
+
+func (s *redisStore) Incr(key string) (int, error) {
+	conn, err := redis.Dial("tcp", s.url)
+	if err != nil {
+		return -1, err
+	}
+	defer conn.Close()
+
+	cnt, err := redis.Int(conn.Do("INCR", key))
+	if err != nil {
+		return -1, err
+	}
+
+	_, err = conn.Do("EXPIRE", key, 59)
```

Comment:
60?

Reply:
I used the example from here: https://redislabs.com/redis-best-practices/basic-rate-limiting/

It seemed correct when I did some manual testing.

Reply:
I did only manual curls, maybe I wouldnt have noticed if you got a "grace" second in the end of each minute.

Reply:
The 59 simply removes the key for the next hour. It should only be retrieved and updated during its minute and needs to be removed before the same minute in the next hour.

Reply:
Ah, I though it meant 59 seconds :)

Reply:
It does mean 59 seconds

Reply:
Same logic just replace minutes with seconds and hour with minute :)

Reply:
maybe put the 59 as a constant and also add that link as a comment

Reply:
I've set the limit to 60. If you did requests on second zero. Then you got new requests in second 59 and then new again in the next second 0. 

Nice catch:)